### PR TITLE
better validation when saving paired fastqs

### PR DESCRIPTION
**Pull request by ryan-williams**
- `logError` for `LENIENT`
- `throw` for `STRICT

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala [hunk 1]
```diff
@@ -419,22 +419,35 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord])
...
+          val readNameOccurrencesMap: collection.Map[Int, Long] = unpairedReadIDsWithCounts.map(_._2).countByValue()
+
+          val msg =
+            List(
+              s"Found $numUnpairedReadIDsWithCounts read names that don't occur exactly twice:",
+
+              readNameOccurrencesMap.map({
+                case (numOccurrences, numReadNames) => s"${numOccurrences}x:\t$numReadNames"
+              }).take(100).mkString("\t", "\n\t", ""),
+              "",
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala]: Nit: if we have more than 100 improperly paired reads, then I would think we should print an ellipsis at the end of the message.

**Comment by fnothaft** [comment]
Other than the small nit, LGTM!

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala (hunk 1 above)
Comment (ryan-williams) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala]: ha yea I meant to add that and forgot, thanks, will add

**Comment by ryan-williams** [comment]
nit should be fixed

**Comment by fnothaft** [comment]
Thanks @ryan-williams! I'll leave this open for comments, but if we don't hear from anyone else tomorrow, I'll merge.

**Comment by fnothaft** [comment]
Thanks @ryan-williams!

**Pull request merged**